packages/runtime/datastore/src/channelStorageService.ts

```diff
@@ -25,7 +26,7 @@ export class ChannelStorageService implements IChannelStorageService {
 
     constructor(
         private readonly tree: ISnapshotTree | undefined,
-        private readonly storage: Pick<IDocumentStorageService, "read">,
+        private readonly storage: Pick<IDocumentStorageService, "read" | "readString">,
```

Comment:
we should only pick readBlob, and optionally readString.
I say optionally, as it's not a requirement (we can synthetize it on the fly as draft PR I just raised demonstrates it). But I prefer for readString to be present on storage interfaces as optional methods, such that if it is there, we call it, but if not - we fill it in (as demoed in my draft PR)